File: InterviewPreparationKit/TwoStrings.py
#!/bin/python3
# https://www.hackerrank.com/challenges/two-strings/problem
import math
import os
import random
import re
import sys

# Complete the twoStrings function below.
def twoStrings(s1, s2):
    chars = set()

    for i in range(len(s1)):
        chars.add(s1[i])

    for j in range(len(s2)):
        if s2[j] in chars:
            return "YES"

    return "NO"

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    q = int(input())

    for q_itr in range(q):
        s1 = input()

        s2 = input()

        result = twoStrings(s1, s2)

        fptr.write(result + '\n')

    fptr.close()

File: Interview Preparation Kit/Dictionaries and Hashmaps/TwoStrings.cpp
// https://www.hackerrank.com/challenges/two-strings/problem
#include <bits/stdc++.h>

using namespace std;

// Complete the twoStrings function below.
string twoStrings(string s1, string s2) {
    unordered_set<char> chars;

    for (int i = 0; i < s1.length(); i++) {
        chars.insert(s1[i]);
    }

    for (int j = 0; j < s2.length(); j++) {
        if (chars.count(s2[j]) > 0) {
            return "YES";
        }
    }

    return "NO";
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    int q;
    cin >> q;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    for (int q_itr = 0; q_itr < q; q_itr++) {
        string s1;
        getline(cin, s1);

        string s2;
        getline(cin, s2);

        string result = twoStrings(s1, s2);

        fout << result << "\n";
    }

    fout.close();

    return 0;
}
